src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,190 @@
+package duke;
+
+import javax.management.Descriptor;
+import java.util.ArrayList;
+
+public class Parser {
+    private User_Input currentType = null;
+    private Ui userInteract;
+    protected TaskList tasks;
+    public boolean isEnd;
+
+
+    public Parser(Ui userInteract, TaskList tasks) {
+        this.userInteract = userInteract;
+        this.tasks = tasks;
+        this.isEnd = false;
+    }
+
+    enum User_Input {
+        TODO,
+        DEADLINE,
+        EVENT,
+        DONE,
+        DELETE,
+        BYE,
+        LIST,
+        FIND
+    }
+
+    /**
+     * Returns a string array that splits the user's command by ' '.
+     * Assigns the user's type of command corresponding to the enum type.
+     *
+     * @param userCommand User's command in a line scanned by Ui.
+     * @return a string array that splits the user's command by ' '.
+     */
+
+    public String[] getDukeType(String userCommand) {
+        String[] words = userCommand.split(" ");
+        if (words[0].equals("deadline")) {
+            this.currentType = User_Input.DEADLINE;
+        } else if (words[0].equals("todo")) {
+            this.currentType = User_Input.TODO;
+        } else if (words[0].equals("event")) {
+            this.currentType = User_Input.EVENT;
+        } else if (words[0].equals("done")) {
+            this.currentType = User_Input.DONE;
+        } else if (words[0].equals("delete")) {
+            this.currentType = User_Input.DELETE;
+        } else if (words[0].equals("bye")) {
+            this.currentType = User_Input.BYE;
+        } else if (words[0].equals("list")) {
+            this.currentType = User_Input.LIST;
+        } else if (words[0].equals("find")) {
+            this.currentType = User_Input.FIND;
+        }
+        return words;
+    }
+
+    /**
+     * Returns a string of response from duke.
+     * The type of response are fixed in Ui.
+     *
+     * @param userCommand User's command scanned by Ui.
+     * @return a string of response from duke.
+     * @throws EmptyInputException,NoResponseException exceptions occur when the user's command cannot be responded or lack of input.
+     */
+
+    public String parse(String userCommand) throws EmptyInputException, NoResponseException {
```

Comment:
Since this method covers a lot of logic, should it be abstract out to different method to improve the readability?